### ISSUE TITLE
Clay DropDown does not focus properly when used with a screen reader.

### ISSUE THREAD

**Issue by wanderlast** [liferay/clay#4994]
Hi! I'm posting this as requested in Slack.

Currently, when using a Clay Dropdown, screen readers cannot correctly 'focus' on the dropdown menu and therefore cannot access the menu with the typical keypresses.

## Is there an example you can provide via codesandbox.com?
You can test it against https://clayui.com/docs/components/drop-down.html or in the Liferay portal with the personal menu.

## What are the steps to reproduce?
When on a page with a dropdown menu created using a Clay DropDown, do the following.

1. Turn on your screen reader.
2. Navigate to the dropdown menu using the tab keys. The screen reader should typically give you the instructions to open the menu using the enter key.
3. Open the menu using the enter key. Press the down arrow to attempt to navigate into the menu.

## What is the expected result?
The screen reader should be able to tell that the menu is 'open' and then proceed to navigate into the menu accordingly.

## Environment

| Tech  | Version |
| ----- | ------- |
| Clay  | v Latest  |
| React |         |

Please see the following conversation for further context and test results: https://liferay.slack.com/archives/CNBG06JS3/p1657929370467129

**Comment by dsanz** [comment]
@matuzalemsteles @marcoscv this issue comes from a support ticket, it'd be great if we could give it some priority
Thank you

**Comment by marcoscv** [comment]
Right, this is happening, at some point we remove the `aria-expanded=""` property to read the state of the dropdown and even if the component works fine at first glance, the reader can't announce the state, so we press enter/space and nothing is read.

This can be fixed by adding `aria-expanded="false"` when the dropdown is not open and ``aria-expanded="true"`` when it is open.
These properties should not be optional, they should always be present.
